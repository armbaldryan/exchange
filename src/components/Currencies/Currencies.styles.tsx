import * as React from "react";

import { styled } from "@mui/material/styles";
import { IconButton, IconButtonProps } from "@mui/material";

export const CurrenciesWrapper = styled("div")(
	() => `
		display: flex;
		flexDirection: row;
		margin: 20px 0;
	`
);

export const SwitchButton = styled((props: IconButtonProps) => (
	<IconButton {...props} />
))(
	() => `
		width: 40px;
		height: 40px;
		align-self: center;
  		margin: 0 20px;
	`
);
