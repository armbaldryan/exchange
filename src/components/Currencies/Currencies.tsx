import React, { SyntheticEvent, useMemo } from "react";

import FormControl from "@mui/material/FormControl";
import Autocomplete from "@mui/material/Autocomplete";
import { CurrenciesWrapper, SwitchButton } from "./Currencies.styles";
import CompareArrowsIcon from "@mui/icons-material/CompareArrows";
import TextField from "@mui/material/TextField";
import { ICurrency } from "store/types";

type Props = {
	currencies: ICurrency;
	from?: string;
	to?: string;
	handleChange: (
		types: "from" | "to"
	) => (event: SyntheticEvent, value: string | null) => void;
	switchHandler: () => void;
};

const Currencies = (props: Props) => {
	const { currencies, from, to, handleChange, switchHandler } = props;

	const currencyOptions = useMemo(() => Object.keys(currencies), [currencies]);

	if (currencies) {
		return (
			<CurrenciesWrapper>
				<FormControl fullWidth>
					<Autocomplete
						onChange={handleChange("from")}
						options={currencyOptions}
						renderInput={(params) => (
							<TextField
								{...params}
								label="From"
								data-testid="from"
								aria-label="from"
							/>
						)}
						value={from}
					/>
				</FormControl>
				<SwitchButton
					disabled={!from || !to}
					aria-label="CompareArrowsIcon"
					color="primary"
					onClick={switchHandler}
				>
					<CompareArrowsIcon />
				</SwitchButton>
				<FormControl fullWidth>
					<Autocomplete
						onChange={handleChange("to")}
						options={currencyOptions}
						renderInput={(params) => (
							<TextField {...params} label="To" data-testid="to" />
						)}
						value={to}
					/>
				</FormControl>
			</CurrenciesWrapper>
		);
	}

	return <p>Loading...</p>;
};

export default Currencies;
