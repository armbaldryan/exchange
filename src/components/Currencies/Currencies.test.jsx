import Currencies from "./Currencies";
import React from "react";
import { render } from "@testing-library/react";

describe("Currencies", () => {
	it("should render Currencies", () => {
		const { container } = render(
			<Currencies
				currencies={{ AMD: "AMD", USD: "USD" }}
				from="AMD"
				to="USD"
				handleChange={jest.fn()}
				switchHandler={jest.fn()}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	it("should render right input values", () => {
		const { getAllByRole } = render(
			<Currencies
				currencies={{ AMD: "AMD", USD: "USD" }}
				from="AMD"
				to="USD"
				handleChange={jest.fn()}
				switchHandler={jest.fn()}
			/>
		);
		const wrapper = getAllByRole("textbox");
		expect(wrapper[0]).toHaveValue("AMD");
		expect(wrapper[1]).toHaveValue("USD");
	});
});
