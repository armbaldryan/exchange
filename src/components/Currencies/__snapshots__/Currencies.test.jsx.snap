// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currencies should render Currencies 1`] = `
<div>
  <div
    class="css-1qwiduc"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-1kpdewa-MuiAutocomplete-root"
        role="combobox"
      >
        <div
          aria-label="from"
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="from"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-378389103"
            id="mui-378389103-label"
          >
            From
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-378389103"
              spellcheck="false"
              type="text"
              value="AMD"
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-186xcr5"
              >
                <span>
                  From
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="CompareArrowsIcon"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-gpy81-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CompareArrowsIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-1kpdewa-MuiAutocomplete-root"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="to"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-231768441"
            id="mui-231768441-label"
          >
            To
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-231768441"
              spellcheck="false"
              type="text"
              value="USD"
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-186xcr5"
              >
                <span>
                  To
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
